mxd/backend-service/src/main/java/org/eclipse/mxd/util/HibernateUtil.java

```diff
@@ -0,0 +1,96 @@
+ * Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *    Contributors:Ravinder Kumar
+ *    Backend-API and implementation
+ *
+ ******************************************************************************/
+
+package org.eclipse.mxd.util;
+
+
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import org.eclipse.mxd.entity.Content;
+import org.eclipse.mxd.entity.Transfer;
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.cfg.Environment;
+import org.hibernate.service.ServiceRegistry;
+
+public class HibernateUtil {
+    private static SessionFactory sessionFactory;
+    private static final Logger logger = Logger.getLogger(HibernateUtil.class.getName());
+
+    public static SessionFactory getSessionFactory() {
+        if (sessionFactory == null) {
+            try {
+                Configuration configuration = new Configuration();
+                Properties settings = new Properties();
+                Properties appProperties = new Properties();
+                try (InputStream input = HibernateUtil.class.getClassLoader().getResourceAsStream("application.properties")) {
+                    appProperties.load(input);
+                } catch (Exception e) {
+                    logger.info(" error " + e.getMessage());
+                }
+                // Read database configuration from environment variables
+                String databaseUrl = System.getenv("BACKEND-SERVICE-DATABASE_URL");
+                String databaseUser = System.getenv("BACKEND-SERVICE-DATABASE_USER");
+                String databasePassword = System.getenv("BACKEND-SERVICE-DATABASE_PASSWORD");
+                // If environment variables are not set, fall back to application properties
+                logger.info(" env databaseUrl : " + databaseUrl);
+                logger.info(" env databaseUser : " + databaseUser);
+                logger.info(" env databasePassword : " + databasePassword);
+                logger.info(" props databaseUrl : " + appProperties.getProperty("database.url"));
```

Comment:
Property key and env key should be same. Env value should take preference over property value.

Reply:
Now I have created SettingResolver class. In SettingResolver and override the properties with Env Variables